shared/src/main/scala/async/futures.scala

```diff
@@ -198,9 +182,9 @@ object Future:
       override def complete(result: Try[T]): Unit = super.complete(result)
 
       override def cancel(): Unit =
-        super.cancel()
-        cancelHandle()
-        reject(CancellationException())
+        if setCancelled() then
+          cancelHandle()
+          reject(CancellationException())
         this.unlink()
```

Comment:
It should work to move this to complete, might be cleaner .. 🤔

Reply:
Moved unlinking to complete